server/spec/consumer_resource_activation_key_spec.rb

```diff
@@ -141,10 +137,9 @@
   it 'should allow a consumer to register with an activation key with an auto-attach across pools' do
     # create extra product/pool to show selectivity
     prod1 = create_product(random_string('product1'), random_string('product1'))
-    subs1 = @cp.create_subscription(@owner['key'], prod1.id, 1)
-    subs2 = @cp.create_subscription(@owner['key'], prod1.id, 1)
-    subs3 = @cp.create_subscription(@owner['key'], prod1.id, 2)
-    @cp.refresh_pools(@owner['key'])
+    create_pool_and_subscription(@owner['key'], prod1.id, 1)
+    create_pool_and_subscription(@owner['key'], prod1.id, 1)
+    create_pool_and_subscription(@owner['key'], prod1.id, 2)
```

Comment:
Refresh pools getting called three times here? Could use the skip flag on the first two.